modules/admin/proc-configuring-postgresql-instance-using-operator.adoc

```diff
@@ -0,0 +1,109 @@
+====
+ 
+
+.Procedure
+
+. Optional: Create a certificate secret to configure your PostgreSQL instance with a TLS connection:
++
+[source,yaml]
+----
+cat <<EOF | oc -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <crt-secret> <1>
+type: Opaque
+stringData:
+ postgres-ca.pem: |-
+  -----BEGIN CERTIFICATE-----
+  <ca-certificate-key> <2>
+ postgres-key.key: |-
+  -----BEGIN CERTIFICATE-----
+  <tls-private-key> <3>
+ postgres-crt.pem: |-    
+  -----BEGIN CERTIFICATE-----
+  <tls-certificate-key> <4>
+  # ...
+EOF
+----
+<1> Provide the name of the certificate secret.
+<2> Provide the CA certificate key.
+<3> Optional: Provide the TLS private key.
+<4> Optional: Provide the TLS certificate key.
+
+. Create a credential secret to connect with the PostgreSQL instance:
++
+[source,yaml]
+----
+cat <<EOF | oc -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <cred-secret> <1>
+type: Opaque
+stringData: <2>
+ POSTGRES_PASSWORD: <password>
+ POSTGRES_PORT: "<db-port>"
+ POSTGRES_USER: <username>
+ POSTGRES_HOST: <db-host>
+ PGSSLMODE: <ssl-mode> # for TLS connection <3>
+ NODE_EXTRA_CA_CERTS: <abs-path-to-pem-file> # for TLS connection, e.g. /opt/app-root/src/postgres-crt.pem <4>
+EOF
+----
+<1> Provide the name of the credential secret.
+<2> Provide credential data to connect with your PostgreSQL instance.
+<3> Optional: Provide the value based on the required link:https://www.postgresql.org/docs/15/libpq-connect.html#LIBPQ-CONNECT-SSLMODE[SSL mode].
+<4> Optional: Provide the value only if you need a TLS connection for your PostgreSQL instance.
+
+. Create a `Backstage` custom resource (CR):
++
+[source,yaml]
+----
+cat <<EOF | oc -n <your-namespace> create -f -
+apiVersion: rhdh.redhat.com/v1alpha1
+kind: Backstage
+metadata:
+  name: <backstage-instance-name>
+spec:
+  database:
+    enableLocalDb: false <1>
+  application: 
+    extraFiles:
+      mountPath: <path> # e g /opt/app-root/src
+      secrets:
+        - name: <crt-secret> <2>
+          key: postgres-crt.pem, postgres-ca.pem, postgres-key.key # key name as in <crt-secret> Secret
+    extraEnvs:
+      secrets:
+        - name: <cred-secret> <3>
+        # ... 
+----
+<1> Set the value of the `enableLocalDb` parameter to `false` to disable creating local PostgreSQL instances.
+<2> Provide the name of the certificate secret if you have configured a TLS connection. 
+<3> Provide the name of the credential secret that you created.
++
+[NOTE]
+====
+The environment variables listed in the `Backstage` CR work with the Operator default configuration. If you have changed the Operator default configuration, you must reconfigure the `Backstage` CR accordingly.
```

Comment:
Please confirm with SME if this is really Backstage CR or Developer Hub CR?

Reply:
@zdrapela, Should we mention  `Developer Hub` CR rather than using `Backstage` CR here? Please confirm.

Reply:
I think we should use the exact CR name. the link mentioned here also point to the `Backstage` CR, not the `Developer Hub` CR. Throughout the doc, we have used the name `Backstage` CR, not `Developer Hub` CR in any code snippet.

Reply:
@kim-tsao ^^?

Reply:
hey folks, to confirm what you see in the UI, you can:
* launch a cluster (eg., send a message to clusterbot on slack `launch 4.16 gcp`
* wait until the cluster spins up
* install the operator subscription from https://github.com/janus-idp/operator/blob/main/.rhdh/docs/installing-ci-builds.adoc
* once the suubscription is created, switch to the Developer perspective, open your chosen namespace, hit the `Add+` button, and scroll down to `Operator-Backed`
* search for developer hub, and when creating/configuring it, note what it says in the YAML view:

I believe you'll see this - even thouigh it's a custom resource for RHDH, the `KIND` is still `Backstage`:

```
kind: Backstage
```

Reply:
The YAML view shows the `Kind` field set to `Backstage` resource.
[
![Backstage-Cr](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/assets/94683525/afa033d9-5224-4f75-9931-9e6202e1bd6a)

Reply:
![image](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/assets/227597/3415f1df-07cb-4381-8300-2265756414b2)